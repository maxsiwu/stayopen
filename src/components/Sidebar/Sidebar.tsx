import React from 'react';
import './Sidebar.css';
import { ILocation } from '../Location/Location';
import { IonList, IonItem, IonLabel, IonContent, IonSearchbar } from '@ionic/react';
import { showFilteredLocations, showLocation } from '../../redux/actions';
import { store } from '../../redux/store';

interface ISidebarProps {
  locations: ILocation[];
}
interface ISidebarState {
  searchText: string;
}
class Sidebar extends React.Component<ISidebarProps, ISidebarState> {
  className = '';
  locationList: ILocation[] = [];

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  constructor (props: any) {
    super(props);
    this.state = {
      searchText: ''
    };
  }

  sendToStore = (value: string): void => {
    const fullList = store.getState().locations;
    store.dispatch(showFilteredLocations(value, fullList));
  };

  openLocation = (location: ILocation): void => {
    store.dispatch(showLocation(location));
  };

  setSearchText = (text: string): void => {
    this.setState({ searchText: text });
  };

  render (): JSX.Element {
    let list = <IonLabel color="medium">Help businesses to stay open</IonLabel>;
    if (this.props.locations && this.props.locations.length > 0 && this.state.searchText !== '') {
      this.className = 'auto';
      list = <IonList lines="none" key="search-result">
        {
          this.props.locations.map((location: ILocation, index: number) => (
            <IonItem key={'search-result-item-' + index} onClick={(): void => this.openLocation(location)} color="light" lines="full" detail={true}>
              <IonLabel color="medium" key={'item-info-' + index}>{location.name}</IonLabel>
            </IonItem>
          ))
        }
      </IonList>;
    }

    if (this.state.searchText && this.state.searchText !== '' && this.props.locations.length === 0) {
      list = <IonLabel color="medium">No result</IonLabel>;
    }

    return (
      <IonContent id="sidebar" key="sidebar-block" className={this.className}>
        <IonSearchbar value={this.state.searchText} color="light" onIonChange={e => {
          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
          this.setSearchText(e.detail.value!);
          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
          this.sendToStore(e.detail.value!);
        }}></IonSearchbar>
        {list}
      </IonContent>
    );
  }
}

export default Sidebar;
