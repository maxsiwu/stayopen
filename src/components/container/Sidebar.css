#sidebar {
    max-width: 320px;
    border: 1px solid rgba(0, 0, 0, 0.1);
    position: fixed;
    z-index: 1;
}
ion-searchbar {
    padding-left: 0 !important;
}