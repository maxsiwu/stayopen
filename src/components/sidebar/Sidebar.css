#sidebar {
    max-width: 320px;
    border: 1px solid rgba(0, 0, 0, 0.1);
    position: fixed;
    z-index: 1;
}
ion-label {
    padding-left: 8px;
}
ion-item {
    cursor: pointer;
}
ion-searchbar {
    --box-shadow: inset 2px 2px 6px 2px rgba(0, 0, 0, 0.1) !important;
}