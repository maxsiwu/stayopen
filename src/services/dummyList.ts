import { ILocation, ICategory, IService } from '../components/Location/Location';

const dummyCategory: ICategory[] = [
  {
    id: 1,
    name: 'Japanese'
  },
  {
    id: 2,
    name: 'Sushi'
  }
];

const dummyService: IService[] = [
  {
    id: 1,
    name: 'Takeout'
  },
  {
    id: 2,
    name: 'Delivery'
  }
];
const dummyList: ILocation[] = [
  {
    id: 1,
    name: 'Miku',
    longitude: 123,
    latitude: 33,
    url: 'www.miku.com',
    services: [dummyCategory[0], dummyCategory[1]],
    categories: [dummyService[0]]
  },
  {
    id: 2,
    name: 'Minami',
    longitude: 145,
    latitude: 23,
    url: 'www.minami.com',
    services: [dummyCategory[0], dummyCategory[1]],
    categories: [dummyService[1]]
  }
];

export default dummyList;
