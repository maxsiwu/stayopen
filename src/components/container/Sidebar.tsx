import React, { useState } from 'react'
import './Sidebar.css'
import { IShop } from '../shop'
import { IonList, IonItem, IonLabel, IonListHeader, IonContent, IonButton, IonSearchbar, IonText } from '@ionic/react'
import { store } from '../..';
import { showFilteredLocations, showShop } from '../actions';

function Sidebar(props: {locations: IShop[]}) {
    const [searchText, setSearchText] = useState('');
    let list = <IonLabel color="medium">Search by typing</IonLabel>;
    if(props.locations && props.locations.length > 0 && searchText !== '') {
        list =  <IonList lines="none" key="search-result">
            { props.locations.map((location, index) => (
                <IonItem key={"search-result-item-" + index} onClick={() => openLocation(location)} color="light" lines="full" detail={true}>
                    <IonLabel color="medium" key={"item-info-" + index}>{location.name}</IonLabel>
                </IonItem>
            ))}
        </IonList>
    }
    if (searchText && searchText !== '' && props.locations.length == 0){
        list = <IonText>No result</IonText>
    }

    return (
        <IonContent id="sidebar" key="sidebar-block" >
            <IonSearchbar value={searchText} onIonChange={e => {
                setSearchText(e.detail.value!);
                sendToStore(e.detail.value!);
            }}></IonSearchbar>
            {/* <IonLabel color="primary" key="sidebar-heading"><h2><b>Search for Businesses</b></h2></IonLabel> */}
            {list}

        </IonContent>
    )
}

const sendToStore = (value: string) => {
    let fullList = store.getState().locations;
    store.dispatch(showFilteredLocations(value, fullList));
}

const openLocation = (location: IShop) => {
    store.dispatch(showShop(location));
}

export default Sidebar