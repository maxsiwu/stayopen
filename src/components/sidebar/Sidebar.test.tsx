import renderer from 'react-test-renderer';
import Sidebar from './Sidebar';
import dummyList from '../../services/dummyList';
import React from 'react';

describe('Counter', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Sidebar locations={dummyList}></Sidebar>).toJSON();
    // const component = renderer.create(Sidebar({ locations: dummyList })).toJSON();

    expect(component).toMatchSnapshot();
  });
});
