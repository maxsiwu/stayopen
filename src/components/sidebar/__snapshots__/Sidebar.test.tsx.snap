// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot renders 1`] = `
<ion-content>
  <ion-searchbar />
  <ion-label>
    Help businesses to stay open
  </ion-label>
</ion-content>
`;
